src/Admin/ActivationRedirect.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Helper class for handling the redirection to the onboarding page.
+ *
+ * @package Automattic\WooCommerce\Pinterest\Admin
+ * @since   X.X.X
+ */
+
+namespace Automattic\WooCommerce\Pinterest\Admin;
+
+use Automattic\WooCommerce\Pinterest\PluginHelper;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class ActivationRedirect
+ */
+class ActivationRedirect {
+
+	use PluginHelper;
+
+	/**
+	 * The setting for redirect to onboarding.
+	 *
+	 * @since  X.X.X
+	 */
+	protected const REDIRECT_SETTING = 'redirect_to_onboarding';
+
+	/**
+	 * Initiate class.
+	 *
+	 * @since  X.X.X
+	 */
+	public function register() {
+		add_action(
+			'admin_init',
+			function () {
+				$this->maybe_redirect_to_onboarding();
+			}
+		);
+	}
+
+	/**
+	 * Checks if merchant should be redirected to the onboarding page.
+	 *
+	 * @since  X.X.X
+	 * @return bool True if the redirection should have happened
+	 */
+	public function maybe_redirect_to_onboarding(): bool {
+		if ( wp_doing_ajax() ) {
+			return false;
+		}
+
+		// If we have redirected before do not attempt to redirect again.
+		if ( ! $this->redirect_setting() ) {
+			return false;
+		}
+
+		// Do not redirect if setup is already complete.
+		if ( Pinterest_For_Woocommerce()::is_setup_complete() ) {
+			$this->update_redirect_setting( false );
+			return false;
+		}
+
+		// Do not redirect if we already are in the Get Started page.
+		if ( $this->is_onboarding_page() && $this->redirect_setting() ) {
+			$this->update_redirect_setting( false );
+			return false;
+		}
+
+		// Redirect if setup is not complete.
+		$this->redirect_to_onboarding_page();
+
+		return true;
+	}
+
+	/**
+	 * Utility function to immediately redirect to the main "Get Started" onboarding page.
+	 * Note that this function immediately ends the execution.
+	 *
+	 * @since  X.X.X
+	 * @return void
+	 */
+	protected function redirect_to_onboarding_page(): void {
+		// If we are already on the onboarding page, do nothing.
+		if ( $this->is_onboarding_page() ) {
+			return;
+		}
+
+		$this->update_redirect_setting( true );
```

Comment:
The current name for the method `update_redirect_setting()` is a bit confusing. I'm noting it here because this is the main place where tracking the setting because a bit hard. Is the setting that we need to do the redirect? Or is it saying that we've already done the redirect?

Reading through this, it looks like the intention is the latter. With that in mind, I think we should update some of the method names to be more descriptive based on what we're actually checking or doing.